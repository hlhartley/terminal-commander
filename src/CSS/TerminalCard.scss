@import '_variables.scss';

.terminal-card {
    align-items: center;
    background-color: black;
    color: $neongreen;
    display: flex;
    flex-flow: column;
    height: 26vh;
    justify-content: center;
    padding: 10px;
    p {
        line-height: 10px;
    }
    input {
        text-align: center;
        margin-bottom: 10px;
        margin-top: 10px;
        width: 380px;
    }
    .score {
        color: $lightred;
    }
    .command-instructions {
        color: $neonblue;
        text-decoration: underline;
    }
    .congrats-message {
        font-size: 35px;
    }
    .fa-flag-usa {
        font-size: 30px;
    }
    .fa-arrow-right {
        margin-right: 10px;
    }
}