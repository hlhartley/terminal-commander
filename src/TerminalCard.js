import React, { Component } from 'react';
import './CSS/main.scss';

export default class TerminalCard extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      score: 2,
      inputValue: '',
    }
  }

  displayFirstCommandDescription() {
    if(!this.props.terminalCommands.length) {
      return 'Reset terminal screen';
    } else {
      return this.props.terminalCommands[`${this.state.score}`].description;
    }
  }

  displayFirstCommandInput() {
    if(!this.props.terminalCommands.length) {
      return 'reset';
    } else {
      return this.props.terminalCommands[`${this.state.score}`].command;
    }
  }

  enterCommand = (event) => {
    this.setState({inputValue: event.target.value})
    }

  updateScore() {
    // if(this.state.inputValue === this.displayFirstCommandInput()) {
    //   this.setState({score: this.state.score++})
      this.setState({score: this.state.score++})
    }
  
  render() {
    return (
        <div className='terminal-card'>
          <p className='mission-text'>Mission: type in the correct commands to get your tank to the target star</p>
          <p>Score: <span className='score'>{this.state.score} / 30</span> complete</p>
          <p>Enter command to "<span className='command-instructions'>{this.displayFirstCommandDescription()}</span>"</p>
          <input placeholder={this.displayFirstCommandInput()} className='command-input' onChange={this.enterCommand} onChange={this.updateScore}></input>
        </div>
      );
    }
  }
