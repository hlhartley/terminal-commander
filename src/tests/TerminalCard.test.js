import React from 'react';
import TerminalCard from '../TerminalCard';
import { shallow } from 'enzyme';

const hhTerminalCommands = [{
  id: 1,
  description: "Reset terminal screen",
  command: "reset"
  },
  {
  id: 2,
  description: "Clear screen",
  command: "clear"
}]

let wrapper;
describe('TerminalCard', () => {

  beforeEach(() => {
    wrapper = shallow(
      <TerminalCard 
        terminalCommands = {hhTerminalCommands}
        score = {0}
        updateScore = { () => {} }
      />
    );
  })

  it('should exist', () => {
    // console.log(wrapper.instance().props)
    expect(wrapper).toBeDefined();
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual({
      inputValue: '',
    })
  })

  it('enterCommand method should set state.inputValue with search string', () => {
    const mockedEvent = {
      target: {
        value: 'mockInput',
      }
    };
    expect(wrapper.state().inputValue).toEqual('');
    wrapper.instance().enterCommand(mockedEvent);
    expect(wrapper.state().inputValue).toEqual('mockInput');
  })
 
  it('clearInput method should change inputValue state to blank string when the enterCommand method is called', () => {
    wrapper.find('.command-input').simulate('click', { inputValue: ''})
    expect(wrapper.state('inputValue')).toEqual('')
  })

  // it('should append the correct number of Terminal Commands to the DOM', () => {
  //   const firstCommand = wrapper.find('ToDontCard').first()
  //   const lastCommand = wrapper.find('ToDontCard').last()

  //   expect(wrapper.find('ToDontCard').length).toEqual(2)
  //   expect firstCommand.props().toDont.title).toEqual('title1')
  //   expect firstCommand.props().toDont.body).toEqual('body1')
  //   expect(lastCommand.props().toDont.title).toEqual('title2')
  //   expect(lastCommand.props().toDont.body).toEqual('body2')
  // })
})

// Remaining methods to test: 
    // displayCommandDescription(), 
  // mock functions being passed as props
  // test 1st if block - possible paths
    // displayCommandInput(),

    // updateScore()
  