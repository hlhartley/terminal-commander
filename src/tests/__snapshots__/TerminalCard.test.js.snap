// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerminalCard should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TerminalCard
    score={0}
    terminalCommands={
      Array [
        Object {
          "command": "reset",
          "description": "Reset terminal screen",
          "id": 1,
        },
        Object {
          "command": "clear",
          "description": "Clear screen",
          "id": 2,
        },
      ]
    }
    updateScore={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="mission-text"
        >
          Mission: type in the correct commands to get your tank to the target star
        </p>,
        <p>
          Score: 
          <span
            className="score"
          >
            0
             / 30
          </span>
           complete
        </p>,
        <p>
          Enter command to "
          <span
            className="command-instructions"
          >
            Reset terminal screen
          </span>
          "
        </p>,
        <input
          className="command-input"
          id="commandInput"
          onChange={[Function]}
          placeholder="reset"
        />,
      ],
      "className": "terminal-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mission: type in the correct commands to get your tank to the target star",
          "className": "mission-text",
        },
        "ref": null,
        "rendered": "Mission: type in the correct commands to get your tank to the target star",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Score: ",
            <span
              className="score"
            >
              0
               / 30
            </span>,
            " complete",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Score: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                0,
                " / 30",
              ],
              "className": "score",
            },
            "ref": null,
            "rendered": Array [
              0,
              " / 30",
            ],
            "type": "span",
          },
          " complete",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Enter command to \\"",
            <span
              className="command-instructions"
            >
              Reset terminal screen
            </span>,
            "\\"",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Enter command to \\"",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset terminal screen",
              "className": "command-instructions",
            },
            "ref": null,
            "rendered": "Reset terminal screen",
            "type": "span",
          },
          "\\"",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "command-input",
          "id": "commandInput",
          "onChange": [Function],
          "placeholder": "reset",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="mission-text"
          >
            Mission: type in the correct commands to get your tank to the target star
          </p>,
          <p>
            Score: 
            <span
              className="score"
            >
              0
               / 30
            </span>
             complete
          </p>,
          <p>
            Enter command to "
            <span
              className="command-instructions"
            >
              Reset terminal screen
            </span>
            "
          </p>,
          <input
            className="command-input"
            id="commandInput"
            onChange={[Function]}
            placeholder="reset"
          />,
        ],
        "className": "terminal-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mission: type in the correct commands to get your tank to the target star",
            "className": "mission-text",
          },
          "ref": null,
          "rendered": "Mission: type in the correct commands to get your tank to the target star",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Score: ",
              <span
                className="score"
              >
                0
                 / 30
              </span>,
              " complete",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Score: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  0,
                  " / 30",
                ],
                "className": "score",
              },
              "ref": null,
              "rendered": Array [
                0,
                " / 30",
              ],
              "type": "span",
            },
            " complete",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Enter command to \\"",
              <span
                className="command-instructions"
              >
                Reset terminal screen
              </span>,
              "\\"",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Enter command to \\"",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reset terminal screen",
                "className": "command-instructions",
              },
              "ref": null,
              "rendered": "Reset terminal screen",
              "type": "span",
            },
            "\\"",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "command-input",
            "id": "commandInput",
            "onChange": [Function],
            "placeholder": "reset",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
